massa-proto/proto/massa/api.proto

```diff
@@ -0,0 +1,302 @@
+syntax = "proto3";
+
+package massa.api.v1;
+
+import "common.proto";
+import "slot.proto";
+
+import "google/api/annotations.proto";
+import "google/rpc/status.proto";
+import "block.proto";
+import "operation.proto";
+
+option cc_enable_arenas = true;
+option csharp_namespace = "Massa.Api.V1";
+option go_package = "google.golang.org/genproto/massa/api/v1/grpc;grpc";
+option java_multiple_files = true;
+option java_package = "com.massa.api.v1";
+option objc_class_prefix = "GRPC";
+option php_namespace = "Massa\\Api\\V1";
+
+// Massa gRPC service
+service Grpc {
+    // GetDatastoreEntries
+    rpc GetDatastoreEntries (GetDatastoreEntriesRequest) returns (GetDatastoreEntriesResponse) {
+        option (google.api.http) = {
+            post: "/v1/get_datastore_entries"
+            body: "*"
+        };
+    }
+
+    // GetSelectorDraws
+    rpc GetSelectorDraws (GetSelectorDrawsRequest) returns (GetSelectorDrawsResponse) {
+        option (google.api.http) = {
+            post: "/v1/get_selector_draws"
+            body: "*"
+        };
+    }
+
+    // GetVersion
+    rpc GetVersion (GetVersionRequest) returns (GetVersionResponse) {
+        option (google.api.http) = {get: "/v1/version"};
+    }
+
+    // GetTransactionsThroughput
+    rpc GetTransactionsThroughput (GetTransactionsThroughputRequest) returns (GetTransactionsThroughputResponse) {
+        option (google.api.http) = {get: "/v1/tx"};
+    }
+
+    // GetNextBlockBestParents
+    rpc GetNextBlockBestParents (GetNextBlockBestParentsRequest) returns (GetNextBlockBestParentsResponse) {
+        option (google.api.http) = {get: "/v1/get_next_block_best_parents"};
+    }
+
+    rpc GetBlocksBySlot (GetBlocksBySlotRequest) returns (GetBlocksBySlotResponse) {
+        option (google.api.http) = {get: "/v1/get_blocks_by_slots"};
+    }
+
+
+
+    // ███████╗████████╗██████╗ ███████╗ █████╗ ███╗   ███╗
+    // ██╔════╝╚══██╔══╝██╔══██╗██╔════╝██╔══██╗████╗ ████║
+    // ███████╗   ██║   ██████╔╝█████╗  ███████║██╔████╔██║
+    // ╚════██║   ██║   ██╔══██╗██╔══╝  ██╔══██║██║╚██╔╝██║
+    // ███████║   ██║   ██║  ██║███████╗██║  ██║██║ ╚═╝ ██║
+
+    // SendBlocks
+    rpc SendBlocks (stream SendBlocksRequest) returns (stream SendBlocksResponse) {}
+
+    // SendEndorsements
+    rpc SendEndorsements (stream SendEndorsementsRequest) returns (stream SendEndorsementsResponse) {}
+
+    // SendOperations
+    rpc SendOperations (stream SendOperationsRequest) returns (stream SendOperationsResponse) {}
```

Comment:
IMHO, we should add a suffix for streams. For example, send_operations could be a one shot call if it's a third party client. But in case of node separation, we have to keep this stream opened.

this comment is valid for all similare cases